docs/references/ic-interface-spec.md

```diff
@@ -7558,40 +7554,34 @@ ic0.cost_cost_http_request<es>(request_size: i64, max_res_bytes: i64, dst: I) :
   copy_cycles_to_canister<es>(dst, arbitrary())
 
 I ∈ {i32, i64}
-ic0.cost_sign_with_ecdsa<es>(src: I, size: I, ecdsa_curve: i32, dst: I) : () = 
-  principal_bytes = copy_from_canister<es>(src, size)
-  if not principal_bytes encode a principal then
-    Trap {cycles_used = es.cycles_used;}
-  if principal_bytes ∉ es.subnets then
-    Trap {cycles_used = es.cycles_used;}
+ic0.cost_sign_with_ecdsa<es>(src: I, size: I, ecdsa_curve: i32, dst: I) : i32 = 
+  key_name = copy_from_canister<es>(src, size)
+  if ecdsa_curve ∉ known_curves then
```

Comment:
maybe say that `known_curves = arbitrary()` and `known_keys = arbitrary()` so that these two identifiers are well-defined; also below